### PR TITLE
Drop support for Rails 5.2, Ruby 2.7, and add support Rails 7.1

### PR DESCRIPTION
**Pull request by stefannibrasil**
It's time.

- drop support to this Rails version: allow us to bump some of the dev dependencies versions.
- add minimum ruby version required to run the repo: `3.0`
- add support for rails 7.1
- add support for ruby 3.3
- fix some deprecation warnings

### CODE REVIEW

File: spec/high_voltage/configuration_spec.rb [hunk 1]
```diff
@@ -6,13 +6,11 @@
 
     before(:each) do
       HighVoltage.configure do |config|
-        ActiveSupport::Deprecation.silence do
```
Comment (stefannibrasil) [inline spec/high_voltage/configuration_spec.rb:9]: `ActiveSupport::Deprecation.silence` is deprecated. I was going to make this work with the new available API but then I realized we don't actually need this block anymore. It was added back when we deprecated caching. It's been a long time -- removing it doesn't even generate any warnings.

**Review by mike-burns — APPROVED** [review]
I'm in favor.

BTW, something I learned while doing this to factory_bot is that people expect this to lead to a major version bump.

**Comment by stefannibrasil** [comment]
Yep! Agreed. There are some unreleased items in the CHANGELOG that could go in this release. I might not have the time to do the release -- will let you know in case you can help.

I'm testing this branch on a new Rails app as well.

**Review by MatheusRich — APPROVED** [review]
Sounds good. Can we release this a release candidate to see if there's anything we missed?

**Comment by stefannibrasil** [comment]
> Sounds good. Can we release this a release candidate to see if there's anything we missed?

That's a great idea! Gonna do that today.

**Pull request merged**